numpy/core/src/multiarray/conversion_utils.c

```diff
@@ -974,7 +986,30 @@ PyArray_IntpFromIndexSequence(PyObject *seq, npy_intp *vals, npy_intp maxvals)
 NPY_NO_EXPORT int
 PyArray_IntpFromSequence(PyObject *seq, npy_intp *vals, int maxvals)
```

Comment:
I wonder if we should deprecate that maxvals thing. Well, not providing it, but using the minimum. If maxvals is smaller, it should really be an error and not just ignore things.

Reply:
If we do that, we'd just deprecate this whole function in favor of the other one.